ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsLocator.java

```diff
@@ -118,6 +123,21 @@
     return filteredHosts;
   }
 
+  @Override
+  public List<List<Struct>> groupByLocation(final List<Struct> keys) {
+    final Map<String, List<Struct>> groups = new HashMap<>();
+    for (Struct key : keys) {
```

Comment:
Here you group keys based on the node that hosts the active partition. However, the keys may belong to different partitions and hence the standby may be different for keys in the same group. I think you should group them per <active, standby> combination else when we do the routing, we cannot assume that all keys go to the same standby  (when active dead)

Reply:
Yeah, I think either grouping by partition or <active, standby> pair would be best.  I think in the normal case where you don't have many standbys, the latter is a good method.